src/Feature/Theme/Feature_Theme.re

```diff
@@ -0,0 +1,101 @@
+open Oni_Core;
+open Revery;
+
+module Log = (val Oni_Core.Log.withNamespace("Oni2.Feature.Theme"));
+
+module Colors = GlobalColors;
+
+type model = {
+  defaults: ColorTheme.Defaults.t,
+  theme: ColorTheme.t,
+};
+
+let defaults =
+  [
+    Colors.Editor.defaults,
+    Colors.List.defaults,
+    Colors.Oni.defaults,
+    Colors.SideBar.defaults,
+    Colors.Tab.defaults,
+    Colors.remaining,
+  ]
+  |> List.map(ColorTheme.Defaults.fromList)
+  |> ColorTheme.Defaults.unionMany;
+
+let initial = contributions => {
+  defaults: ColorTheme.Defaults.unionMany([defaults, ...contributions]),
+  theme: ColorTheme.{variant: Dark, colors: ColorTheme.Colors.empty},
+};
+
+let fallback = keyName => {
+  Log.warnf(m => m("Fallback color used for: %s", keyName));
+  Revery.Colors.black; // TODO: Transparent? Magenta?
+};
```

Comment:
To avoid having to deal with an `option` in the case of a missing value, we use a fallback function that provides some stupid default. vscode will just return `undefined` when a color is missing. Unsure what does during rendering though, but I suspect the effect is much the same as what this fallback function provides.

Reply:
I like the idea of using Magenta here - to make it very explicit when we miss one.

Reply:
Magenta it is then!

Reply:
I expect some great memes will come from this.